cfy_manager/components/statusreporter/status_reporter.py

```diff
@@ -0,0 +1,69 @@
+#########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from ...config import config
+from ...utils.common import sudo
+from ...logger import get_logger
+from ...utils.systemd import systemd
+from ..base_component import BaseComponent
+from ..components_constants import SOURCES
+from ...utils.install import yum_install, yum_remove
+
+STATUS_REPORTER = 'statusreporter'
+
+logger = get_logger('status-reporter')
+
+
+class StatusReporter(BaseComponent):
+    def __init__(self, skip_installation, reporter_type):
+        super(StatusReporter, self).__init__(skip_installation)
+        self.reporter_type = reporter_type
+
+    def _build_extra_config_flags(self):
+        return ''
+
+    def _should_install(self):
+        return True
+
+    def install(self):
+        if not self._should_install():
+            logger.notice('Not installing Status Reporter while not'
+                          ' in external setup...')
+            return
+        logger.notice('Installing Status Reporter for {0}...'.format(self.reporter_type))
+        yum_install(config[STATUS_REPORTER][SOURCES]['cloudify_status_reporter_url'])
+        logger.notice('Status Reporter successfully installed')
+
+    def configure(self):
+        if not self._should_install():
+            # Nothing to config
+            return
+        logger.notice('Configuring Status Reporter...')
+        config[STATUS_REPORTER]['reporter_type'] = self.reporter_type
+        config[STATUS_REPORTER]['extra_config_flags'] = self._build_extra_config_flags()
```

Comment:
this sucks, lets not add things to config, especially if they're gonna be overwritten several times

instead i think add `**kwargs` or a `additional_render_context={}` to `systemd.configure`